### DISCUSSION TITLE
Is there a world where ReactFlow can work with 1000+ nodes/edges?

### DISCUSSION THREAD

**Discussion by CamboimGabriel** [xyflow/xyflow#3003] [General]
When try to implement a miro/whimsical clone, this is a question that keeps in my mind and seems like mandatory for this type of solution

**Comment by bcakmakoglu** [comment]
As far as I know React Flow is not intended to be used in that kind of scale (really depends on the complexity of your nodes though).
A canvas based approach would be better for that sort of use-case as the performance would be a lot better though there's trade-offs using a canvas instead.

**Comment by 4096void** [comment]
at this moment, I tried 200+ nodes, with 4k+ edges the browser just blackout
